MariaDB/server
# MDEV-27889 Make table cache mutex contention threshold configurable

- [x] *The Jira issue number for this PR is: MDEV-27889*


## Description
Adding two new config values:
    TABLE_OPEN_CACHE_K_QUERIES_PER_SEC: This is the expected throughput
         in thousands of queries per second.
         Default: 100
    TABLE_OPEN_CACHE_INSTANCE_THRESHOLD: The threshold before an
         instance is considered contended in percentage of
         table_open_cache_k_queries_per_sec
         Default: 20

The hard coded values in the code before this were based on values from
an Intel Broadwell system (which has been discontinued for 4 years already).
The default values maintain this old value.

Allowing these to be configurable future proofs the code slightly.

For comparison, on a Cascade Lake system, we see steady state reached in 16 minutes with unmodified code and 6 seconds with TABLE_OPEN_CACHE_INSTANCE_THRESHOLD set to 10 (hard coded is equivalent to setting this to be the default value of 20)

## How can this PR be tested?

While the automated testing will see the new parameters exist, to properly test this you need to run a stress test that gets table cache contention, for example HammerDB. 


## Basing the PR against the correct MariaDB version


## Backward compatibility
This maintains backward compatibility.
Older versions are not forward compatible if the new parameters are set in the config file because the older version will see unrecognized parameters in the config file and error out.
